http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -153,6 +155,9 @@ private HttpRequestBase wrapEntity(HttpExecuteRequest request,
             if (!request.httpRequest().firstMatchingHeader(HttpHeaders.CONTENT_LENGTH).isPresent()) {
                 entity = ApacheUtils.newBufferedHttpEntity(entity);
             }
+            if (request.httpRequest().matchingHeaders("Transfer-Encoding").contains("chunked")) {
+                entity = new ApacheChunkedHttpEntity(entity);
```

Comment:
Do we need this new class `ApacheChunkedHttpEntity`? Can we just pass isChunked to `RepeatableInputStreamRequestEntity`?

Reply:
We need the class since BufferedHttpEntity always returns false for isChunked(), regardless of the underlying entity

Reply:
I see, can we do something like the following?
```
if (!request.httpRequest().firstMatchingHeader(HttpHeaders.CONTENT_LENGTH).isPresent() && notChunked) {
   entity = new ApacheChunkedHttpEntity(entity); 
}

Reply:
updated and setting `chunked` to true in `RepeatableInputStreamRequestEntity`